kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala

```diff
@@ -315,8 +315,10 @@ abstract class EngineYarnModeSubmitter extends Logging {
       }
       // respect the following priority loading configuration, and distinct files
       // hive configuration -> hadoop configuration -> yarn configuration
-      val hiveConf = kyuubiConf.getOption(KYUUBI_ENGINE_DEPLOY_YARN_MODE_HIVE_CONF_KEY)
-      listDistinctFiles(hiveConf.get).foreach(putEntry)
+      if ("hive".equalsIgnoreCase(engineType)) {
```

Comment:
For this class, we want it to be as generic as possible. Can it be better abstracted? This would allow different subclasses to implement varying logic.